src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -299,25 +299,43 @@ private function process_directives_args( string $html, array &$context_stack, a
 
 			/*
 			 * Sorts the attributes by the order of the `directives_processor` array
-			 * and checks what directives are present in this element. The processing
-			 * order is reversed for tag closers.
+			 * and checks what directives are present in this element.
 			 */
-			$directives_prefixes = array_intersect(
```

Comment:
Would it be possible to introduce a new private method:

```php
private function is_exiting_tag() {
    return $p->is_tag_closer() || ! $p->has_and_visits_its_closer_tag();
}
```
and put it in the place where `$p->is_tag_closer()` call was used before?

Reply:
I thought of using something like that. The problem that when `! $p->has_and_visits_its_closer_tag()` is true, we want to run the directives in "entering" mode AND in "exiting" mode, so I think we can't assume that `is_exiting_tag` is always true in that case.

Reply:
I got it now. It wasn't immediately obvious when reading the code. What would help in the long run would be to define in the constructor the list of directives executed in the entering mode and the list of directives executed in the exiting mode. This way, all these operations we see today:

```php
$directive_processor_prefixes          = array_keys( self::$directive_processors );
$directive_processor_prefixes_reversed = array_reverse( $directive_processor_prefixes );
```

wouldn't have to be repeated for every processed tag.